clippy_lints/src/needless_pass_by_value.rs

```diff
@@ -286,7 +286,7 @@ impl<'tcx> LateLintPass<'tcx> for NeedlessPassByValue {
                         input.span,
                         "consider taking a reference instead",
                         format!("{}&{}", snippet(cx, before_span, "_"), snippet(cx, after_span, "_")),
-                        Applicability::MaybeIncorrect,
+                        Applicability::Unspecified,
```

Comment:
> Mark the suggestion as `Applicability::Unspecified` as the proposed
fix will not compile since we suggest modifying the parameter type
without modifying the arguments at the call sites.

This is one of the use cases `MaybeIncorrect` is for: on its own definitely incorrect. So `MaybeIncorrect` is the correct applicability. This overload of `MaybeIncorrect` is one motivation for https://github.com/rust-lang/cargo/issues/13028

Reply:
I've repushed without the applicability change commit.